src/main/java/server/request/RequestParser.java

```diff
@@ -1,41 +1,48 @@
 package server.request;
 
 import java.util.Arrays;
+import java.util.HashMap;
 import server.constants.Format;
+import server.constants.Header;
 import server.constants.Method;
 import server.constants.Path;
 
 public class RequestParser {
-  private String rawRequest;
+  private final String rawRequest;
 
-  public Request parse(String rawRequest) {
+  public RequestParser(String rawRequest) {
     this.rawRequest = rawRequest;
-
-    return new Request(method(), path(), body());
-  }
-
-  private String startLine() {
-    return rawRequest.split(Format.BREAK)[0];
   }
 
-  private Method method() {
+  public Method method() {
     String stringMethod = startLine().split(" ")[0];
     return Method.get(stringMethod);
   }
 
-  private Path path() {
+  public Path path() {
     String stringPath = startLine().split(" ")[1];
     return Path.get(stringPath);
   }
 
-  private String body() {
-    String[] splitBody = rawRequest.split(Format.DBL_BREAK);
+  public HashMap<Header, String> headers() {
+    String requestHead = rawRequest.split(Format.DBL_BREAK)[0];
+    String[] headLines = requestHead.split(Format.BREAK);
+
+    HashMap<Header, String> mappedHeaders = new HashMap<>();
+    String[] headers = Arrays.copyOfRange(headLines, 1, headLines.length);
 
-    if (splitBody.length > 1) {
-      String[] bodyLines = Arrays.copyOfRange(splitBody, 1, splitBody.length);
-      return String.join(Format.DBL_BREAK, bodyLines);
-    } else {
-      return null;
+    for (String line : headers) {
+      if (!line.isBlank()) {
+        String[] lineData = line.split(": ");
+        Header headerType = Header.get(lineData[0]);
+        mappedHeaders.put(headerType, lineData[1]);
```

Comment:
What is going to happen if you received unsupported header?

Reply:
I just tested this and saw that the request fails. Do you think it would make more sense to either 1) add all standard request header types to the Enum and configure this method to ignore non-standard headers or, 2) store Header as a String type instead of Enum?

Reply:
I think storing this as a string will be the easiest way to do this, at least for request. I don't remember if you need to use them in future. Let's not overcomplicate this.

Reply:
Ok will do! Thanks for the input